library/core/tests/iter/traits/iterator.rs

```diff
@@ -616,6 +616,65 @@ fn test_next_chunk() {
     let mut it = std::iter::repeat_with(|| panic!());
     assert_eq!(it.next_chunk::<0>().unwrap(), []);
 }
+#[test]
+fn test_happy_path_item_not_in_iterator() {
+    assert!(![1i32, 2i32, 3i32].iter().contains(&4i32));
+}
+
+#[test]
+fn test_edge_case_handling_none_values() {
+    assert!([Some(2i32), Option::<i32>::None].iter().contains(&None));
+    assert!([Some(2i32), Option::<i32>::None].iter().contains(&Some(2i32)));
+}
+
+#[test]
+fn test_edge_case_handling_empty_iterator() {
+    assert!(!Vec::<i32>::new().iter().contains(&1i32));
+}
+
+#[test]
+fn test_edge_case_handling_iterator_with_duplicates() {
+    assert!([1i32, 2i32, 2i32, 3i32].iter().contains(&2i32));
+}
+
+#[test]
+/// Tests that short-circuiting works correctly when using `contains`
+/// When you run the function, it should move the iterator forward after the first appearance of the item
+fn test_short_circuiting() {
+    let vector: Vec<i32> = vec![1i32, 2i32, 3i32, 1i32, 1i32];
+    let mut iterator = vector.into_iter();
+    assert!(iterator.contains(1i32));
+    assert_eq!(iterator.next(), Some(2));
+    assert!(!iterator.contains(4i32));
+    assert_eq!(iterator.next(), None);
+}
+
+#[test]
+fn test_edge_case_handling_iterator_with_custom_struct() {
```

Comment:
Nit: since these tests only check `.contains()`, they should have `test_contains` in the name